addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/predefined/utility/Transliterate.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.predefined.utility;
+
+import java.io.IOException;
+import java.text.Normalizer;
+import org.zaproxy.addon.encoder.processors.predefined.DefaultEncodeDecodeProcessor;
+
+public class Transliterate extends DefaultEncodeDecodeProcessor {
+
+    private static final Transliterate INSTANCE = new Transliterate();
+
+    @Override
+    protected String processInternal(String value) throws IOException {
+        // Normalize with compatible decomposition, then remove anything non-ASCII
+        return Normalizer.normalize(value, Normalizer.Form.NFKD).replaceAll("[^\\p{ASCII}]", "");
```

Comment:
Could also consider using the Apache commons util `StringUtils#stripAccents` (if it's already a dependency): https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/StringUtils.html#stripAccents-java.lang.String-

Reply:
I did come across that when I was considering this but I didn't see any benefit vs the native impl.

Reply:
Looking at that code they are doing things slightly different, e.g. use of `InCombiningDiacriticalMarks`. It would also probably be better to call it something else since this not actually transliterating but...